collect_app/src/main/java/org/odk/collect/android/activities/SplashScreenActivity.java

```diff
@@ -231,7 +231,13 @@ public void run() {
     }
 
     private void initMapBox() {
-        MapView mapView = findViewById(R.id.mapView);
-        mapView.getMapAsync(mapBoxMap -> mapBoxMap.setStyle(Style.MAPBOX_STREETS, style -> { }));
+        // This "one weird trick" lets us initialize MapBox at app start when the internet is
+        // most likely to be available. This is annoyingly needed for offline tiles to work.
+        try {
+            MapView mapView = getLayoutInflater().inflate(R.layout.mapbox_init, null).findViewById(R.id.mapView);
+            mapView.getMapAsync(mapBoxMap -> mapBoxMap.setStyle(Style.MAPBOX_STREETS, style -> { }));
+        } catch (Exception ignored) {
```

Comment:
I would catch Error here as well it might be needed in some cases like https://github.com/getodk/collect/commit/f6bc94ebef3d901734b272d8e5031abd90245bd0

Reply:
Good call! We want to account for everything.